### PR TITLE
Improve conference management

### LINKED ISSUE #25: Meeting Join button will not be disabled again
Once a user has entered a meeting, the button with Join meeting is always displayed for the lecturers/tutors, even if they have already left the conference or it no longer exists.

Steps to Reproduce:

- Enter class room with stuents
- (Private tab) Enter class room with lecturers
- Enter conference with students (Top right)
- End conference
- (Private Tab) Join button is still next to username

### LINKED ISSUE #27: No invitation sent, when inviter is in conference.

### LINKED ISSUE #35: Conference window opens multiple times
When joining, leaving and rejoining a conference, the web-gui opens multiple tabs for succinct joins.

### LINKED ISSUE #36: Conference not found in Backend
Steps to reproduce:

- Open conference via button.
- Leave conference.
- Wait for deletion in Backend.
- Reopen Conference
- Exception in Server -> NoSuchElementException

### LINKED ISSUE #37: Invitations sent automatically.
After an invitation was sent before, any new conference of the inviting user will trigger an invitation.

### LINKED ISSUE #46: Manage multiple conferences per user.
To make invitations behave more predictable, it is preferable to manage mutliple conferences per user in backend and frontend.

### LINKED ISSUE #64: Delete own tickets
Everybody should have the possibility to delete his own tickets. For example, if a question of a student has been already resolved.

### PR DESCRIPTION
**Pull request by snicki13**
Fixes #25 
Fixes #27 
Fixes #35 
Fixes #36
Fixes #37 
Fixes #46
Fixes #64

### CODE REVIEW

**Review by mxsph — CHANGES_REQUESTED** [review]
During testing I noticed a couple of errors:

- [ ] When you get an invitation, you get the following error message:  `Uncaught (in promise): NotSupportedError: The media resource indicated by the src attribute or assigned media provider object was not suitable.`. This should not be displayed because the browser either does not support this function or the user has rejected the authorization (Thus this error is intentional). Maybe you could add a setting later to activate this function (ask for permission again).
- [ ] The button Make conference invisible does not work.
- [ ] The Exit Conference button does not work. Although the user is removed from the conference in the `digital-classroom`, this user does not actually leave/end the conference. Additionally, this error is displayed: `TypeError: t is undefined `
- [x] The conference list also displays conferences that have already been closed/terminated. Is this intentional? In general I like the idea to show the conferences, later we should probably add better management options here (like ending the conference).
- [x] It looks like it will fix #25 as well?

> Conferences are displayed in frontend (still ugly though)

:+1: Since there are other parts of this application whose design should be adapted in my opinion, I have created an issue for this (#55).

**Comment by snicki13** [comment]
The remaining "issues" existed before this PR and will be fixed on another branch within a web-ui overhaul.

**Comment by snicki13** [comment]
Let's go!

File: web-gui/src/app/page-components/classroom/conferences/conferences.component.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  styleUrls: ['./conferences.component.scss']
+})
+export class ConferencesComponent {
+
+  @Input() conferences: ConferenceInfo[]
+  @Input() currentUser: UserDisplay
+
+  constructor() {
+    console.log("conferences")
+  }
```
Comment (mxsph) [inline web-gui/src/app/page-components/classroom/conferences/conferences.component.ts:17]:
```suggestion
```

File: web-gui/src/app/page-components/menu-bar/menu-bar.component.ts [hunk 2]
```diff
@@ -13,16 +20,28 @@ import {AuthService} from "../../service/auth.service";
...
+
+  }
+
+  public createConference() {
+      this.dialog.open(CreateConferenceDialogComponent, {
+        height: 'auto',
+        width: 'auto',
+        data: new CreateConferenceInputData(this.classroomService.classroomInfo, this.classroomService.currentUser)
+      }).beforeClosed().pipe(
+        tap(_ => console.log('kek')),
```
Comment (mxsph) [inline web-gui/src/app/page-components/menu-bar/menu-bar.component.ts:40]:
```suggestion
```

**Review by mxsph — CHANGES_REQUESTED** [review]
It is no longer possible to invite someone to a conference: 

- User: When trying to invite a user to a conference this error is displayed `No ticket or conference provided for invitation!`.
- Ticket: When trying to invite a user to a conference no error is shown but the creator of the ticket does not get an invitation.

**Review by mxsph — APPROVED** [review]

**Pull request merged**